pallene/parser.lua

```diff
@@ -577,25 +592,30 @@ function Parser:FuncParams()
 end
 
 function Parser:FuncExp(func_token)
-    local params = self:FuncParams()
-    local rt_types = {}
-    if self:peek(":") then
-        self:region_begin()
-        self:e()
-        rt_types = self:RetTypes()
-        self:region_end()
-    end
+    local oparen = self:e("(")
+    local params = {}
 
-    local block    = self:Block()
-    local _        = self:e("end", func_token)
+    if self:peek("NAME") then
+        repeat
+            local id = self:e("NAME")
+            if self:peek(":") then
+                local colon = self:e()
+                self:syntax_error(colon.loc, "Function expressions cannot be type annotated")
```

Comment:
Would it be possible to implement this using `FuncParams()`? Parse the function parameters and then loop over the result to check if the type is present or not (similarly to what we did in the FuncStat).

Reply:
Oh, yeah that's a great idea. done :+1: